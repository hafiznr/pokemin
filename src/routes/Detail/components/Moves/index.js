import { Box, Skeleton, Tag } from "@chakra-ui/react";

const Moves = ({
  loading,
  moves
}) => {
  return (
    <div>
      <Box fontSize="2xl" mb="4px">Moves</Box>
      {loading ? (
        <Skeleton 
          height="100px" 
          data-testid="moves-skeleton"
        />
      ) : (
        <Box 
          display="flex" 
          flexWrap="wrap"
          overflow="scroll"
          height="100px"
          bgColor="gray.100"
          borderRadius="8px"
          borderColor="gray.100"
          borderWidth="4px"
          data-testid="moves-container"
        >
          {moves?.map((item, i) => (
            <Tag 
              key={i}
              size="lg"
              margin="0 4px 4px 0"
              bgColor="white"
            >
              {item.move.name}
            </Tag>
          ))}
        </Box>
      )}
    </div>
  )
}

export default Moves;