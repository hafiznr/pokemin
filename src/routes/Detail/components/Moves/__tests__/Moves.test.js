import { render, screen } from '@testing-library/react';
import Moves from '../';

test('renders Moves', () => {
  const { container, rerender } = render(<Moves/>);

  expect(container).toMatchSnapshot();

  rerender(<Moves loading/>);
  expect(screen.getByTestId('moves-skeleton')).toBeInTheDocument();

  const moves = [
    {
      move: {
        name: 'swords-dance'
      }
    },
    {
      move: {
        name: 'leaf-storm'
      }
    }
  ];

  rerender(<Moves moves={moves}/>);
  expect(screen.getByTestId('moves-container'))
    .toHaveTextContent('swords-danceleaf-storm');
});