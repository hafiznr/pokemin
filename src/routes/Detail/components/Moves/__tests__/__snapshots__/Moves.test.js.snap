// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Moves 1`] = `
<div>
  <div>
    <div
      class="css-bn3r0t"
    >
      Moves
    </div>
    <div
      class="css-8ehw8a"
      data-testid="moves-container"
    />
  </div>
</div>
`;
